src/stores/HomeStore.js

```diff
@@ -328,37 +328,42 @@ class HomeStore {
     try {
       const homeValidatorsAddress = await this.homeBridge.methods.validatorContract().call()
       this.homeBridgeValidators = new this.homeWeb3.eth.Contract(BRIDGE_VALIDATORS_ABI, homeValidatorsAddress);
-      this.validators = await getBridgeValidators(this.homeBridgeValidators)
+
       this.requiredSignatures = await this.homeBridgeValidators.methods.requiredSignatures().call()
       this.validatorsCount = await this.homeBridgeValidators.methods.validatorCount().call()
 
-      if(this.validators.length !== Number(this.validatorsCount)) {
-        this.homeBridgeValidators = new this.homeWeb3.eth.Contract(REWARDABLE_BRIDGE_VALIDATORS_ABI, homeValidatorsAddress);
-        this.validators = await getBridgeValidators(this.homeBridgeValidators)
-      }
+      this.validators = await getValidatorList(homeValidatorsAddress, this.homeWeb3.eth)
     } catch(e){
       console.error(e)
     }
   }
 
-
   async getStatistics() {
     try {
       const events = await getPastEvents(this.homeBridge, 0, 'latest')
-      this.processLargeArrayAsync(events, this.processEvent)
+      this.processLargeArrayAsync(events, this.processEvent('UserRequestForSignature', 'AffirmationCompleted'))
+
+      // if contracts started with V1 version we should get that statistics too
+      const homeBridgeV1 = new this.homeWeb3.eth.Contract(HomeBridgeV1Abi, this.HOME_BRIDGE_ADDRESS);
```

Comment:
Will it save time if we do this only in case if V1 contracts were deployed previously?

Reply:
Yes, it will save resources if we avoid doing one extra call. The only issue I see is how to detect that V1 contracts were deployed previously

Reply:
Is it possible to catch `Upgraded` events and find if `getBridgeInterfacesVersion` called for the corresponding implementation will cause error since there is no such method in the contract? My understanding is this is only the condition when the contract a) was upgraded b) upgraded from v1. What do you think? Should we make this filter even more precise?

Reply:
I think that is a good strategy to detect that a contract was deployed using v1 contracts but it makes me think if it is worth applying it here. We will need to perform two requests to get this information (one for `Upgraded` events and one for `getBridgeInterfacesVersion`). Although on most cases contracts will be from v2 version and we will ended up doing this two requests only (3 requests if it is v1), I think it's better the code as it is, with just one request since it will require less time and resources to get the list of validators displayed. Does it makes sense to you?